pkgs/development/python-modules/gpib-ctypes/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, pytestCheckHook
+, pytest-runner
+, linux-gpib
+}:
+
+buildPythonPackage rec {
+  pname = "gpib_ctypes";
+  version = "0.3.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-c9l6TNmM4PtbvopnnFi5R1dQ9o3MI39BHHHPSGqfjCY=";
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-runner
+  ];
+
+  postPatch = ''
+    substituteInPlace gpib_ctypes/gpib/gpib.py \
+      --replace "libnames = [filename] if filename else ['libgpib.so.0', 'gpib-32.so']" \
+      "libnames = [filename] if filename else ['${linux-gpib}/lib/libgpib.so.0', 'libgpib.so.0', 'gpib-32.so']"
```

There are other ways, such as using `os.add_dll_directory`, but I think this is OK.

```suggestion
    substituteInPlace gpib_ctypes/gpib/gpib.py \
      --replace "libgpib.so.0" "${linux-gpib}/lib/libgpib.so.0"
```
I think it could be simpler.